actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -514,6 +516,14 @@ def resolve_image_source(source, skip_pipeline)
           raise ArgumentError, "Can't resolve image into URL: #{e}"
         end
 
+        def resize_image_source(source, options)
+          return source unless resize_active_storage_images
+          return source unless options[:width] || options[:height]
+          return source unless source.is_a?(ActiveStorage::Attached) || source.is_a?(ActiveStorage::Attachment) || source.is_a?(ActiveStorage::Blob)
```

Comment:
Is it possible to use duck-typing here? I think `source&.variable?` could work and also prevents us trying to resize something like an SVG

Reply:
The safe navigator was not enough since the source could be a string or a variant, neither of which respond to variable?

Reply:
Oh, but I’ve changed to use `variable?`